test/byContract/IexecEscrow/IexecEscrowToken.test.ts

```diff
@@ -0,0 +1,457 @@
+
+const standardAmount = AmountWithDecimals(BigNumber.from(100));
+const depositParams = {
+    amount: standardAmount,
+};
+const depositArgs = Object.values(depositParams) as [BigNumber];
+const withdrawParams = {
+    amount: standardAmount,
+};
+const withdrawArgs = Object.values(withdrawParams) as [BigNumber];
+
+describe('IexecEscrowTokenDelegate', () => {
+    let proxyAddress: string;
+    let [iexecPoco, , iexecPocoAsAccountA, iexecPocoAsAdmin]: IexecInterfaceToken[] = [];
+    let [iexecAdmin, accountA, accountB, anyone]: SignerWithAddress[] = [];
+    let [rlcInstance, rlcInstanceAsAccountA]: RLC[] = [];
+
+    beforeEach('Deploy', async () => {
+        proxyAddress = await loadHardhatFixtureDeployment();
+        await loadFixture(initFixture);
+    });
+
+    async function initFixture() {
+        const accounts = await getIexecAccounts();
+        ({ iexecAdmin, anyone: accountA, requester: accountB, anyone } = accounts);
+
+        iexecPoco = IexecInterfaceToken__factory.connect(proxyAddress, anyone);
+        iexecPocoAsAccountA = iexecPoco.connect(accountA);
+        iexecPocoAsAdmin = iexecPoco.connect(iexecAdmin);
+        rlcInstance = await RLC__factory.connect(await iexecPoco.token(), anyone);
+        rlcInstanceAsAccountA = rlcInstance.connect(accountA);
+        await rlcInstance
+            .connect(iexecAdmin)
+            .transfer(accountA.address, AmountWithDecimals(BigNumber.from(10_000)))
+            .then((tx) => tx.wait());
+    }
+
+    describe('Receive and Fallback', () => {
+        it('Should revert on receive', async () => {
+            await expect(
+                accountA.sendTransaction({
+                    to: iexecPoco.address,
+                    value: standardAmount,
+                }),
+            ).to.be.revertedWith('fallback-disabled');
+        });
+        it('Should revert on fallback', async () => {
+            const randomData = ethers.utils.hexlify(
+                ethers.utils.toUtf8Bytes((Math.random() * 0xfffff).toString(16)),
+            );
+            await expect(
+                accountA.sendTransaction({
+                    to: iexecPoco.address,
+                    value: standardAmount,
+                    data: randomData,
+                }),
+            ).to.be.revertedWith('fallback-disabled');
+        });
+    });
+
+    describe('Deposit', () => {
+        it('Should deposit tokens', async () => {
+            await rlcInstanceAsAccountA
+                .approve(iexecPoco.address, depositParams.amount)
+                .then((tx) => tx.wait());
+
+            expect(await iexecPocoAsAccountA.callStatic.deposit(...depositArgs)).to.be.true;
+            await expect(iexecPocoAsAccountA.deposit(...depositArgs))
+                .to.changeTokenBalances(
+                    rlcInstance,
+                    [accountA, iexecPoco],
+                    [-depositParams.amount, depositParams.amount],
```

Comment:
Maybe we should check
- token balances on RLC **and** sRLC
- event on sRLC as you do